examples/petsc/multigrid.c

```diff
@@ -216,7 +216,7 @@ int main(int argc, char **argv) {
     ierr = DMSetVecType(dm[i], vec_type); CHKERRQ(ierr);
     PetscInt dim;
     ierr = DMGetDimension(dm[i], &dim); CHKERRQ(ierr);
-    ierr = SetupDMByDegree(dm[i], level_degrees[i], num_comp_u, dim,
+    ierr = SetupDMByDegree(dm[i], level_degrees[i], q_extra, num_comp_u, dim,
```

Comment:
How are you keeping track of the fine level quadrature order in here?

Reply:
Do we use multigrid for `bp2`?

Reply:
Not in `./bps`,but in `./multigrid` we certainly can.

I just don't see how the quadrature orders would agree here.

Reply:
Yeah, I didn't touch multigrid in this PR, in the older one I modified this line.
I wanted to narrow down the issue, so in this PR we only have new `basis_x` and `basis_u` (compare with main) and I cannot solve a simple projection problem.

Reply:
Hmm, this is basis_x in this PR
```
CeedBasis (H^1 space on a hexahedron element): dim=3 P=2 Q=3
      qref1d:    -0.77459667      0.00000000      0.77459667
   qweight1d:     0.55555556      0.88888889      0.55555556
    interp1d[0]:          0.88729833      0.11270167
    interp1d[1]:          0.50000000      0.50000000
    interp1d[2]:          0.11270167      0.88729833
      grad1d[0]:         -0.50000000      0.50000000
      grad1d[1]:         -0.50000000      0.50000000
      grad1d[2]:         -0.50000000      0.50000000
```

in main
```
CeedBasis (H^1 space on a hexahedron element): dim=3 P=2 Q=5
      qref1d:    -0.90617985     -0.53846931      0.00000000      0.53846931      0.90617985
   qweight1d:     0.23692689      0.47862867      0.56888889      0.47862867      0.23692689
    interp1d[0]:          0.95308992      0.04691008
    interp1d[1]:          0.76923466      0.23076534
    interp1d[2]:          0.50000000      0.50000000
    interp1d[3]:          0.23076534      0.76923466
    interp1d[4]:          0.04691008      0.95308992
      grad1d[0]:         -0.50000000      0.50000000
      grad1d[1]:         -0.50000000      0.50000000
      grad1d[2]:         -0.50000000      0.50000000
      grad1d[3]:         -0.50000000      0.50000000
      grad1d[4]:         -0.50000000      0.50000000
```

Reply:
Oh, sometimes PETSc mangles the initial value of a option if the option is not found

Reply:
If I manually set `-q_extra 1`, I seem to get the same results as main

Reply:
```diff
$ git diff
diff --git a/examples/petsc/bps.c b/examples/petsc/bps.c
index c97cc9e1..8e9b9f4c 100644
--- a/examples/petsc/bps.c
+++ b/examples/petsc/bps.c
@@ -414,7 +414,7 @@ static PetscErrorCode Run(RunParams rp, PetscInt num_resources,
     // Create DM
     PetscInt dim;
     ierr = DMGetDimension(dm_deg, &dim); CHKERRQ(ierr);
-    ierr = SetupDMByDegree(dm_deg, rp->degree, q_extra, rp->num_comp_u, dim,
+    ierr = SetupDMByDegree(dm_deg, rp->degree, rp->q_extra, rp->num_comp_u, dim,
                            bp_options[rp->bp_choice].enforce_bc,
                            bp_options[rp->bp_choice].bc_func); CHKERRQ(ierr);
     for (PetscInt r = 0; r < num_resources; r++) {
```

Reply:
That's weird, q_extra is defined few lines above and it is equal to `rp->q_extra`